server.js

```diff
@@ -149,15 +150,30 @@ LimitdServer.prototype.start = function (done) {
   return this;
 };
 
-LimitdServer.prototype.stop = function () {
+LimitdServer.prototype.stop = function (callback) {
   var self = this;
   var log = self._logger;
   var address = self._server.address();
+  callback = cb(callback || _.noop).timeout(5000).once();
+  log.debug(address, 'closing server');
 
-  this._server.destroy(function() {
+  this._server.destroy((err) => {
+    if (err) {
+      log.error({ err, address }, 'error closing the tcp server');
+      return callback(err);
```

Comment:
don't we want to close the db either way in the case? we failed to close the server, but there is no reason for the db closing to also fail?